$MirOS: ports/devel/libgtop/patches/patch-libgtop-sysdeps_m4,v 1.1.7.1 2005/03/18 15:44:51 tg Exp $
--- libgtop-sysdeps.m4.orig	Tue Jan  2 23:50:08 2007
+++ libgtop-sysdeps.m4	Tue Jun 26 22:57:29 2007
@@ -67,7 +67,7 @@ AC_DEFUN([GNOME_LIBGTOP_SYSDEPS],[
 	  libgtop_have_sysinfo=yes
 	  libgtop_need_server=no
 	  ;;
-	freebsd*|netbsd*|openbsd*|bsdi*|kfreebsd*)
+	freebsd*|netbsd*|openbsd*|bsdi*|kfreebsd*|mirbsd*)
 	  libgtop_sysdeps_dir=freebsd
 	  libgtop_use_machine_h=yes
 	  libgtop_need_server=yes
@@ -150,7 +150,7 @@ AC_DEFUN([GNOME_LIBGTOP_SYSDEPS],[
 #include <net/netisr.h>
 #include <net/route.h>
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__FreeBSD_kernel__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__FreeBSD_kernel__) || defined(__OpenBSD__)
 #include <net/if_sppp.h>
 #else
 #include <i4b/sppp/if_sppp.h>
@@ -176,7 +176,7 @@ AC_DEFUN([GNOME_LIBGTOP_SYSDEPS],[
 #include <net/netisr.h>
 #include <net/route.h>
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__FreeBSD_kernel__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__FreeBSD_kernel__) || defined(__OpenBSD__)
 #include <net/if_sppp.h>
 #else
 #include <i4b/sppp/if_sppp.h>
@@ -203,6 +203,7 @@ AC_DEFUN([GNOME_LIBGTOP_SYSDEPS],[
 	    msginfo_needs=
 	    for def in nothing KERNEL _KERNEL; do
 	      AC_COMPILE_IFELSE([#define $def
+#include <sys/types.h>
 #include <sys/ipc.h>
 #include <sys/msg.h>
 #include <stdio.h>
